### PR TITLE
Add basic test for Enki::Network module.

### PR DESCRIPTION
**Pull request by mthl**
Let the executables in the 'tests' directory use the thread library.  This is
done by moving the 'find_package' call for "Threads" in top level
"CMakelists.txt".

### CODE REVIEW

**Comment by mthl** [comment]
@sebpouteau let me know if there is no issue with this test on MacOS.

**Comment by sebpouteau** [comment]
I have already written this test.  
This test have a same problem what mine if you add a many connection for exemple:

```c 
for (int i = 1; i < 10; i++)
            {
                Client* c = new Client(SERVER_IP);
                std::thread threadClient([c]() { c->run(); });
                threadClient.detach();

                THEN( "the server has a connection" ) {
                    REQUIRE( s->getConnectionNumbers() == i );
                }
```
Sometimes this test failed.

Please, Look my branch tmp-network !

**Comment by mthl** [comment]
> This test have a same problem what mine if you add a many connection

The question was about this particular not an extension of it.

> Please, Look my branch tmp-network !

It is not convenient for me to work on the network tests if you are not sharing what you are doing incrementally with small PRs.  

Can you either send review this PR or send your own soon?  Thanks

**Comment by mthl** [comment]
I have updated my patch by using:
- the `join` method for threads
- object on the stack to benefit from automatic deletion once out of scope.

**Comment by mthl** [comment]
the added test fails on windows with the following error:
```
11/11 Test #11: network_basic ....................***Exception: Other  0.42 sec

91% tests passed, 1 tests failed out of 11
```
This output is somewhat terse to be useful.  @davidjsherman do you if this is possible to ssh into the Jenkins windows node to debug this?

**Comment by davidjsherman** [comment]
If I run the test manually, I see that the shared library for Dashel cannot be found:

> ```
> ci@INRIACI-OT9HJCJ MINGW32 /c/builds/workspace/PDP_network-basic-test-QBHKY3BRXY
> CKS5ACRXXQ53532GYARDQUIC3ARIQU7RRZN52LJJRQ/build/windows/tests
> $ ./network.exe "A basic connection"
> C:/builds/workspace/PDP_network-basic-test-QBHKY3BRXYCKS5ACRXXQ53532GYARDQUIC3AR
> IQU7RRZN52LJJRQ/build/windows/tests/network.exe: error while loading shared libr
> aries: libdashel.dll: cannot open shared object file: No such file or directory
> ```

**Comment by sebpouteau** [comment]
problem solved.

File: tests/network.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+    sleep(1);
+    return tmp < TIME_OUT ;
+}
+
+void create_client(Client *c)
+{
+    c = new Client("127.0.0.1");
+}
+using namespace Enki;
```
Comment (Krazyxx) [inline tests/network.cpp]: Duplication of code

File: tests/network.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+void threadClient(Client* c){
+    c->run();
+}
+
+bool check(Server * s, int nbConnection){
+    sleep(1);
+    clock_t t = clock();
+    clock_t tmp;
+    while((tmp = ( clock() - t)) < TIME_OUT && s->getConnectionNumbers() != nbConnection){
+        usleep(1000);
```
Comment (nicpalard) [inline tests/network.cpp]: isn't usleep(1000) the same as sleep(1) ?

File: tests/network.cpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+            REQUIRE( check(server, 1) );
+            
+            delete client;
+            
+            server->stop();
+            thread_server.join();
+            delete server;
+            
+        }
+        WHEN( "2 clients" )
```
Comment (nicpalard) [inline tests/network.cpp]: Isn't this part useless ? It is already tested in the `WHEN ("10 Clients") `test case.

**Pull request merged**